### PR TITLE
Removed code duplication in `server.js`

### PR DESCRIPTION
**Pull request by wojta**
## Motivation
Code is being duplicated in `server.js`.

## What
Improve a litle bit the code, so it doesn't have the practically same `newExpressApp()` and `newApolloServer()` two times in there.

## Why
Improve readability.

## How
Create `newServer()` function that covers new server and app creation and is called from proper places.

## Checklist:

- [x] Code has been tested locally by PR requester
- [ ] Changes have been successfully verified by another team member

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -95,7 +109,7 @@ class DataSyncServer {
     if (this.server) await this.server.close()
   }
 
-  async onReceive () {
+  async onSchemaRebuild () {
```
[the remark this answers is not in the document]
Comment (wojta) [reply server/server.js]: ok

File: server/server.js [hunk 2]
```diff
@@ -83,9 +97,9 @@ class DataSyncServer {
     this.server.startListening = startListening.bind(this.server)
 
     // Initialize the schema listener for hot reload
-    this.schemaListener = schemaListenerCreator(schemaListenerConfig)
-    this.debouncedOnReceive = _.debounce(this.onReceive, 500).bind(this)
-    this.schemaListener.start(this.debouncedOnReceive)
+    this.schemaListener = schemaListenerCreator(this.serverConfig.schemaListenerConfig)
```
[the remark this answers is not in the document]
Comment (wojta) [reply server/server.js]: moved back to `initialize()`

File: server/server.js [hunk 3]
```diff
@@ -83,9 +86,20 @@ class DataSyncServer {
...
+    this.debouncedOnSchemaRebuild = _.debounce(this.onSchemaChangedNotification, 500).bind(this)
+    this.schemaListener.start(this.debouncedOnSchemaRebuild)
+  }
+
+  /**
+   * Starts or restarts express app and Apollo server
+   * @param {boolean} restartApp if true, restarts also Express app
```
Comment (darahayes) [inline server/server.js]: This seems like it's not meant to be here 🤔 Perhaps it was from an earlier implementation. If you can remove that then we're good to go.

**Comment by wtrocki** [comment]
@wojta This PR broke authorization feature. @darahayes fixed that on #91

**Pull request merged**